### ISSUE TITLE
Code splitting with react isn't working

### ISSUE THREAD

**Issue by egoist** [inertiajs/inertia#696]
### Versions:

- `@inertiajs/inertia` version: 0.9.0
- `@inertiajs/inertia-react` version: 0.6.0

### Describe the problem:

[Using `import()` in resolveComponent](https://inertiajs.com/client-side-setup#code-splitting) won't work, results in following error in browser:

```
React.createElement: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: object. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.
```

I believe this is due to the `component` here is a `Promise` object:

https://github.com/inertiajs/inertia/blob/88bf77eb1f5406d416351532f6deb7b7ea1448c7/packages/inertia-react/src/App.js#L13

### Steps to reproduce:

 Initialize a minimal React / Laravel project, adding following code to `app.js`:

```js
import { App } from '@inertiajs/inertia-react'
import React from 'react'
import { render } from 'react-dom'

const el = document.getElementById('app')

render(
  <App
    initialPage={JSON.parse(el.dataset.page)}
    resolveComponent={(name) =>
      import(`./Pages/${name}`).then((res) => res.default)
    }
  />,
  el,
)
```

Adding some pages in `Pages` folder, and run the server.

**Comment by reinink** [comment]
Thanks for sharing this. We've made some major changes in these (the latest) releases of Inertia and the adapters, so I'm not surprised a bug slipped through. These releases are brand new (from yesterday), and I haven't even had a chance to document them yet.

That said, I've been able to reproduce this, and am working on a fix. 👍

**Comment by reinink** [comment]
This has been fixed as part of #697. 👍
